UnitTesting/ProcedureUnitTests.cs

```diff
@@ -0,0 +1,147 @@
+            list.AddPoint((Convert.ToDateTime("2000/9/1")), 0.0);
+            list.AddPoint((Convert.ToDateTime("2000/10/1")), 0.0);
+            list.AddPoint((Convert.ToDateTime("2000/11/1")), 0.0);
+            list.AddPoint((Convert.ToDateTime("2000/12/1")), 0.0);
+            return list;
+        }
+
+        public List<IGrouping<int, SegmentEventsEntry>> GetTestDataMonthly()
+        {
+            var lst = new List<SegmentEventsEntry>();
+            lst.Add(GetSegmentObject("leak", 120, 0, 1, "1/1/2019 12:04:00 PM"));
+            lst.Add(GetSegmentObject("leak", 240, 120, 1, "1 / 1 / 2019 12:10:15 PM"));
+            lst.Add(GetSegmentObject("leak", 120, 0, 2, "1/1/2019 12:20:0 PM"));
+            lst.Add(GetSegmentObject("leak", 240, 120, 2, "1/1/2019 12:50:15 PM"));
+            lst.Add(GetSegmentObject("leak", 600, 300, 1, "2/1/2019 13:10:15 PM"));
+            lst.Add(GetSegmentObject("leak", 120, 60, 3, "2/1/2019 13:20:00 PM"));
+            lst.Add(GetSegmentObject("leak", 240, 0, 1, "2/1/2019 13:40:00 PM"));
+            lst.Add(GetSegmentObject("leak", 60, 0, 3, "3/1/2019 14:05:15 PM"));
+            lst.Add(GetSegmentObject("leak", 60, 0, 1, "3/1/2019 14:10:15 PM"));
+            lst.Add(GetSegmentObject("leak", 120, 0, 1, "3/1/2019 14:55:15 PM"));
+            return lst.GroupBy(a => a.TimeStamp.Hour).ToList();
+        }
+
+        public DataPoints<DateTime,double> GetTestExpectedDaily()
+        {
+            //Build data
+            var list = new DataPoints<DateTime, double>();
+            list.AddPoint((Convert.ToDateTime("1/1/2019 12:04:00 PM")),8.0);
+            list.AddPoint((Convert.ToDateTime("1/1/2019 13:10:15 PM")), 10.0);
+            list.AddPoint((Convert.ToDateTime("1/1/2019 14:05:15 PM")), 4.0);
+
+            return list;
+        }
+
+        public List<IGrouping<int, SegmentEventsEntry>> GetTestDataDaily()
+        {
+            var lst = new List<SegmentEventsEntry>();
+            lst.Add(GetSegmentObject("leak", 120, 0, 1, "1/1/2019 12:04:00 PM"));
+            lst.Add(GetSegmentObject("leak", 240, 120, 1, "1 / 1 / 2019 12:10:15 PM"));
+            lst.Add(GetSegmentObject("leak", 120, 0, 2, "1/1/2019 12:20:0 PM"));
+            lst.Add(GetSegmentObject("leak", 240, 120, 2, "1/1/2019 12:50:15 PM"));
+            lst.Add(GetSegmentObject("leak", 600, 300, 1, "1/1/2019 13:10:15 PM"));
+            lst.Add(GetSegmentObject("leak", 120, 60, 3, "1/1/2019 13:20:00 PM"));
+            lst.Add(GetSegmentObject("leak", 240, 0, 1, "1/1/2019 13:40:00 PM"));
+            lst.Add(GetSegmentObject("leak", 60, 0, 3, "1/1/2019 14:05:15 PM"));
+            lst.Add(GetSegmentObject("leak", 60, 0, 1, "1/1/2019 14:10:15 PM"));
+            lst.Add(GetSegmentObject("leak", 120, 0, 1, "1/1/2019 14:55:15 PM"));
+            return lst.GroupBy(a => a.TimeStamp.Hour).ToList();    
+        }
+
+        public SegmentEventsEntry GetSegmentObject(string eve,double inn, double outt,int segid,string time)
```

Comment:
I get you were trying to avoid using the same name as the keywords, but why not use the same names as the `SegementEventsEntry` object?

```c#
GetSegmentObject(string eventType, double flowIn, double flowOut, int segmentId, string timestamp)
```

Reply:
Hi this was an error on my part. Please excuse me. I have renamed it;